test/petab/test_petab.py

```diff
@@ -1,13 +1,173 @@
+import sys
+import os
+import numpy as np
+import scipy.stats
+import pandas as pd
+import pytest
+import git
+import itertools
+
 import amici.petab_import
 import petab
+import petab.C as C
 import pyabc.petab
+import pyabc.petab.base
 
-import git
-import os
-import numpy as np
+
+@pytest.fixture(params=itertools.product(
+    [petab.C.LIN, petab.C.LOG, petab.C.LOG10],
+    [*petab.C.PRIOR_TYPES, None]))
+def prior_specs(request):
+    """A one-line parameter df for a given prior type."""
+    scale, prior_type = request.param
+    var1, var2 = 0.2, 0.9
+    if prior_type:
+        # dataframe with objective prior
+        df = pd.DataFrame({
+            C.PARAMETER_ID: ['p1'],
+            C.ESTIMATE: [1],
+            C.PARAMETER_SCALE: [scale],
+            C.LOWER_BOUND: [np.nan],
+            C.UPPER_BOUND: [np.nan],
+            C.OBJECTIVE_PRIOR_TYPE: [prior_type],
+            C.OBJECTIVE_PRIOR_PARAMETERS: [f"{var1};{var2}"],
+        })
+    else:
+        # also consider the case that no prior is specified, resulting in a
+        #  parameter scale uniform prior within the rescaled bounds
+
+        # unscale variables
+        unscaled_var1, unscaled_var2 = var1, var2
+        if scale == C.LOG:
+            unscaled_var1, unscaled_var2 = np.exp([var1, var2])
+        elif scale == C.LOG10:
+            unscaled_var1, unscaled_var2 = 10.**var1, 10.**var2
+        # dataframe without objective prior
+        df = pd.DataFrame({
+            C.PARAMETER_ID: ['p1'],
+            C.ESTIMATE: [1],
+            C.PARAMETER_SCALE: [scale],
+            C.LOWER_BOUND: [unscaled_var1],
+            C.UPPER_BOUND: [unscaled_var2],
+        })
+        # expected default if objective type not set
+        prior_type = C.PARAMETER_SCALE_UNIFORM
+    yield scale, prior_type, var1, var2, df
+
+
+def test_petab_prior(prior_specs):
+    """Test whether the prior is correctly defined by sampling from it."""
+    # need to fix random seed due to stochastics of multiple testing
+    np.random.seed(0)
+
+    # extract settings
+    scale, prior_type, var1, var2, parameter_df = prior_specs
+
+    # create prior from petab data frame
+    pyabc_prior = pyabc.petab.base.create_prior(parameter_df)
+
+    # generate random samples
+    n_samples = 10000
+    samples = pyabc_prior.rvs(size=n_samples)['p1']
+
+    # -- UNIFORM COVERAGE -- #
+
+    # check that uniform parameters fill their domain
+    if prior_type in [C.UNIFORM, C.PARAMETER_SCALE_UNIFORM]:
+        assert (samples >= var1).all() and (samples <= var2).all()
+        assert ((samples >= var2 - (var2-var1)*0.01).any() and
+                (samples <= var1 + (var2-var1)*0.01).any())
+
+    # -- MEAN AND VARIANCE -- #
+
+    # sample mean and variance
+    mean = np.mean(samples)
+    var = np.var(samples)
+
+    # ground truth mean and variance
+    if prior_type in [C.UNIFORM, C.PARAMETER_SCALE_UNIFORM]:
+        mean_th = var1 + (var2 - var1) / 2
+        var_th = (var2 - var1)**2 / 12
```

Comment:
Where does the `12` come from?

Reply:
that's the variance of  (a,b) uniforma variable https://en.wikipedia.org/wiki/Continuous_uniform_distribution

Reply:
I guess one should know, thanks :see_no_evil: